.github/workflows/build_workflow.yml

```diff
@@ -67,6 +85,7 @@ jobs:
 
   publish-docs:
     if: ${{ github.event_name == 'push' }}
+    needs: [pre-commit-hooks, build]
```

Comment:
You should remove this line because `pre-commit-hooks` and `build` can be skipped via `check-jobs-to-skip`. If `publish-docs` is dependent on those jobs, it might not run after merging the PR to `main` if they are skipped.

`check-jobs-to-skip` is included in the e3sm diag build workflow, but not in zstash. I think you mixed both workflow files a bit.